onnxruntime/core/graph/graph.cc

```diff
@@ -2409,7 +2414,28 @@ void Graph::CleanUnusedInitializers() {
   }
 
   std::for_each(erase_list.cbegin(), erase_list.cend(),
-                [this](const std::string& name) { name_to_initial_tensor_.erase(name); });
+                [this](const std::string& name) {
+                  RemoveInitializedTensor(name);
+
+                  // handle edge case where the unused initializer has a matching graph input
+                  auto& proto_inputs = *graph_proto_->mutable_input();
+                  auto i = std::find_if(proto_inputs.begin(), proto_inputs.end(),
+                                        [&name](const ONNX_NAMESPACE::ValueInfoProto& input) {
+                                          return input.name() == name;
+                                        });
+
+                  if (i != proto_inputs.end()) {
+                    RemoveRepeatedFieldEntry(proto_inputs, i);
```

Comment:
>RemoveRepeatedFieldEntry(proto_inputs, i) [](start = 20, length = 41)

For a extreme edge case, could there be an initializer being both graph input and output, without any computation node?

Reply:
There could, but it would be used in that case wouldn't it? 

---
In reply to: [344001779](https://github.com/microsoft/onnxruntime/pull/2320#discussion_r344001779) [](ancestors = 344001779)

Reply:
OK seems handled above.

---
In reply to: [344002365](https://github.com/microsoft/onnxruntime/pull/2320#discussion_r344002365) [](ancestors = 344002365,344001779)